docs/usage/configuration/config_documentation.md

```diff
@@ -462,6 +462,21 @@ See the docs [request log format](../administration/request_log.md).
 * `additional_resources`: Only valid for an 'http' listener. A map of
    additional endpoints which should be loaded via dynamic modules.
 
+  _Added in Synapse 1.86.0_
+
+* UNIX socket support:
+  * `path`: A path and filename for a UNIX socket. Ensure it is in a readable/writeable
+     directory and is pre-existing(the directory will not be created). No default
+    * **Note**: Declaring a `path` is not compatible while also having a `port`
+       configuration for the same `listener`
+    * **Note**: `metrics` and `manhole` are not supported as `types` on UNIX socket `listeners`
+    * The `x_forwarded` option will default to true and may be left out.
+    * Other options that would not make sense to use with a UNIX socket, such as 
+    `bind_address[es]` and `tls` will be ignored and can be removed.
+  * `mode`: The file permissions to set on the UNIX socket. Defaults to `666`
+  * `type: http` is valid (provided `metrics` is not included in the `resources`:`names`)
```

Comment:
```suggestion
UNIX socket support (_Added in Synapse 1.86.0_):

* `path`: A path and filename for a Unix socket. Make sure it is located in a directory with read and write permissions, and that it already exists (the directory will not be created). Defaults to `None`."
   * **Note**: Declaring a `path` is not compatible while also having a `port`
       configuration for the same `listener`
   * The `x_forwarded` option defaults to true  when using Unix sockets and can be omitted.
   * Other options that would not make sense to use with a UNIX socket, such as 
    `bind_addresses` and `tls` will be ignored and can be removed.
* `mode`: The file permissions to set on the Unix socket (must be at least TODO). Defaults to `666`
* **Note:** Must be set as `type: http` (does not support `metrics` and `manhole`). Also make sure that `metrics` is not included in `resources` -> `names`
```

Reply:
Some of that doesn't feel right...But I like most of it(Me bad at words ☹️).

The sub-heading of `Added in Synapse 1.86.0`, I would like to keep that just above to maintain consistency with the rest of the document(which I think was mentioned in another PR, I can find if you are interested). However, I do believe a good bolding would be nice(I tried it out in another comment chain [here](https://github.com/matrix-org/synapse/pull/15708#discussion_r1231571110)). Also need to change that to 1.87.0 now(and anywhere else it's mentioned).

`path` doesn't default to `None`, it has no default. `None` is an explicit value that is empty of meaning, so that's not right.

Remove the `(must be at least TODO)` from the `mode` line. (That was settled, right?)

The indentation changed a bit, was that intentional?

Mind if I tweak it using your suggestion but with those edits?

Reply:
> `path` doesn't default to `None`, it has no default. `None` is an explicit value that is empty of meaning, so that's not right.

Looks like it defaults to `None`:

https://github.com/matrix-org/synapse/blob/2ac6c3bbb535677bd62b3df425dd1755dba79b66/synapse/config/server.py#L942


> Remove the `(must be at least TODO)` from the `mode` line. (That was settled, right?)

We can remove the `mode` stuff if the option is being removed



> The indentation changed a bit, was that intentional?

Yes as it didn't make sense to me to group "UNIX socket support" as another list item. And can just be more details like how "Valid resource names are:" is.


> Mind if I tweak it using your suggestion but with those edits?

Up to you if you think that's better.

Reply:
I really don't want to remove `mode` as a configurable. I almost included the only other option I could find(it was `wantPID` which controls if a lock file is created with the PID). I defaulted that to `True` as some light reading suggested it would make `systemd` and such systems happy, and was harmless for docker images/containers.

Reply:
Perhaps if it's documented in such a way as to be explicit it's for special circumstances with use by a security professional, and therefore just omit it for a standard installation?

Reply:
> Looks like it defaults to `None`:
> 
> https://github.com/matrix-org/synapse/blob/2ac6c3bbb535677bd62b3df425dd1755dba79b66/synapse/config/server.py#L942

Yeah, but look at the rest:
https://github.com/matrix-org/synapse/blob/0f02f0b4da92229e88e27a92ea3bfa523457bfc1/synapse/config/server.py#L942-L958

None is not allowed. Gosh this looks messy, in retrospect.

> > The indentation changed a bit, was that intentional?
> 
> Yes as it didn't make sense to me to group "UNIX socket support" as another list item. And can just be more details like how "Valid resource names are:" is.

I think I see what you mean by this, after staring at it for a while. Let me see what I can do about that with everything else.

Reply:
> None is not allowed. Gosh this looks messy, in retrospect.

I don't see anything there that doesn't allow `None`. You just have to have `port` or `socket_path`.

In any case, this is really about how we want to explain defaults like this. Just make a decision and we can further nit from there if needed.

Reply:
It's simple enough. We say there is no default. Programmatically, yes it will default to `None` which means it would have to be non-existent in the yaml, or left empty. It's considered a 'special' value in this context. However, if it *is* `None` and there is no `port` defined as an alternative, then the `ConfigError` raises telling them they need to pick one. We don't choose a default for this(not only because every system is different) as it would fall to that automatically and maybe they just misconfigured a `port`(which incidentally also doesn't really have a default either)

Reply:
Updated in [39597fc](https://github.com/matrix-org/synapse/pull/15708/commits/39597fcf0bb13d41964cf400a5b6d06992eec926)

(except this most current part about no defaults)